### DISCUSSION TITLE
In what conditions next-auth may create multiple version of cookies for different domains?

### DISCUSSION THREAD

**Discussion by krzysztof-dk** [nextauthjs/next-auth#10313] [Help]
Hi guys. I'm working on a project using next-auth and Keycloak and facing a behaviour of having multiple cookies set for different domains as a result of oAuth login handled by next-auth.

The issue I'm facing because of that is that due to having this much cookies I'm hitting the cookie size limit of server providing me an images used in HTML <a> tag, namely I'm getting `431 Request Header Fields Too Large` error when browser tries to download the image through the HTML.

The question I have are:
1. In what conditions next-auth will create multiple version of cookies for different domains?
2. Can I tune/alert that?
3. Is this a bug/miss behaviour?
4. Could you point me to the next-auth source code part responsible for this behaviour?

<img width="1937" alt="next-auth-cookies-debug" src="https://github.com/nextauthjs/next-auth/assets/106629516/0b5dea85-2f9d-476f-acc0-67a34bb27d30">

The screenshot shows situation when I'm running project from localhost but this is the same after deployment.

**Comment by krzysztof-dk** [comment]
Example from deployment
<img width="2557" alt="next-auth-cookies-debug" src="https://github.com/nextauthjs/next-auth/assets/106629516/7f5449d7-baac-40e5-ac40-a826449d16e3">
 env

**Comment by krzysztof-dk** [comment]
It seems that second set of cookies is created due to having `issuer` (https://next-auth.js.org/providers/keycloak) set to different domain than the one the page itself is using.

**Reply by benlimanto-salt** [reply]
@krzysztof-dk 
You should look into 

https://stackoverflow.com/a/78405808/4906348
 
https://next-auth.js.org/configuration/options

Cookies in NextAuth.js are chunked by default, meaning that once they
reach the 4kb limit, we will create a new cookie with the .{number}
suffix and reassemble the cookies in the correct order when parsing /
reading them. This was introduced to avoid size constraints which can
occur when users want to store additional data in their sessionToken,
for example.

**Comment by guillerg01** [comment]
If you get here, the problem is that too much user information is being saved in the session, and it seems that all of this is saved in the access token, so check that the things you save are not too large.
